docs/source/service_revisions.rst

```diff
@@ -0,0 +1,91 @@
+i.e. service revisions can be registered and requested from it.
+
+Registering a service revision
+==============================
+To register a service revision:
+
+.. code-block:: python
+
+    import requests
+
+    response = requests.post(
+        "<base_url>/<chosen_path_for_django_twined_urls>/services/<namespace>/<name>",
+        json={"revision_tag": "<revision_tag>", "is_default": True},
+    )
+
+For example, if your base URL is ``myapp.org/api``, you've registered the ``django-twined`` URLs under
+``integrations/octue``, and the service revision you want to register is ``my-org/my-service:1.2.9``, the request would
+be:
+
+.. code-block:: python
+
+    import requests
+
+    response = requests.post(
+        "myapp.org/api/integrations/octue/services/my-org/my-app",
+        json={"revision_tag": "1.2.9", "is_default": True},
+    )
+
+Getting a specific service revision
+===================================
+Make a request like the following:
+
+.. code-block:: python
+
+    import requests
+
+    response = requests.get(
+        "myapp.org/api/integrations/octue/services/my-org/my-app?revision_tag=1.2.9",
+    )
+
+    response.json()
+    >>> {
+        "namespace": "my-org",
+        "name": "my-service",
+        "revision_tag": "1.2.9",
+        "is_default": True,
+    }
+
+Currently, the only useful information this provides is whether the requested service revision is the default or not.
+Later, more useful information will be returned including how to send a question to that specific service revision and
+access tokens to do so.
+
+Getting the latest service revision
+===================================
+This request is the same as above except the revision tag is omitted. By default, the service revision with the latest
+semantic version revision tag will be returned.
+
+.. code-block:: python
+
+    import requests
+
+    response = requests.get(
+        "myapp.org/api/integrations/octue/services/my-org/my-app",
+    )
+
+    response.json()
+    >>> {
+        "namespace": "my-org",
+        "name": "my-service",
+        "revision_tag": "1.2.9",
+        "is_default": True,
+    }
+
+Controlling which service revision is returned by default
+=========================================================
+The ``TWINED_SERVICE_REVISION_SELECTION_CALLBACK`` setting can be set to a user-defined callable to control which
+service revision is returned when a service is requested but the revision tag isn't specified. This callable must take
+two keyword arguments: ``namespace`` and ``name`` and must return a single instance of the ``ServiceRevision`` model.
+
+Examples of how this feature can be used include:
+
+- A/B testing
+- Returning certain service revisions based on the request or the requester
+- Controlling the availability of beta versions of services
+- Other custom routing of questions to services
```

Comment:
OK here's what we need to do about is_default.

There are two different patterns we could adopt:

**OPTION 1 - resolve the latest on the fly (ie at request time)**
In this case, the default revision tag is determined by executing the callback on each request to the `services/my-org/my-service` endpoint. `is_default` is then a derived value that should not be stored.

**OPTION 2 - resolve the latest at registration time**
If not None, the callback should be executed prior to save of the ServiceRevision model (ie before [this line](https://github.com/octue/django-twined/blob/a426e55874e41905db221f3438f65f12a67b1a99/django_twined/models/service_revisions.py#L188)). The callback should receive the new (as-yet-unsaved) instance (because the callback will need the revision tag) and return a boolean to be assigned to the `is_default` value on the model.

If None, the `is_default` value is left as-is on the instance and it's the server admins' responsibility to set it as they see fit.

Perhaps the callback could be called `TWINED_SERVICE_REVISION_IS_DEFAULT_CALLBACK`.

In this case, the `is_default` value  needs to stay as a concrete column in the database, essentially to store the result of the callback.

The default callback might do "if the tag matches a non-candidate release regex then promote to default otherwise leave the value as-is".

**Conclusion**
I favour option 2. If the callback is computationally intensive (which it could be, pulling out a bunch of rows from a database, parsing all their revision tags and choosing one) then running it only on creation (very infrequently) could be a major performance benefit compared to running it inline with the question asking process.

The drawback of option 2 is that you can't do anything stochastic there, or anything that depends on the time of request - but I can't really think of any use cases for that. If we really needed to we could add another callback to achieve that later.

**Note** I carefully said "leave as-is" rather than "set false" because you'll want to allow the admin to override the value in save.